exploration/text-vs-code.md

```diff
@@ -136,16 +209,93 @@ The current syntax includes some plain-ascii keywords:
+the general consensus that formed for simple messages
+is that we allow them to be unquoted.
+
+("Simple messages" refers to messages consisting solely of a pattern, and thus are not complex messages.)
+
+Because the simple message pattern consists of the entire message,
+the pattern includes any leading or trailing whitespace.
+
+Given simple messages already being decided at a high level,
+the design decisions below for the proposed and alternative designs pertain specifically to complex messages.
 
 ## Proposed Design
 
+### Start in text, encapsulate message, always quote patterns
+
+Description:
+
+Since simple messages are unquoted (starting in text mode),
+complex messages must also start in text mode.
+
+Within a complex message, patterns are always quoted with `{{...}}` or other choice of delimiter.
+
+The entire complex message is also wrapped with `{{...}}` or other choice of delimiter.
+This allows interior "code mode" of message to have flexible whitespace in between tokens
+and _around_ quoted patterns.
+
+Pros
+
+* The rule about the whether leading and trailing whitespace is included is simple and unambiguous.
+* This matches the WYSIWIG behavior that simple messages preserve.
+* The patterns can be detected within the pattern more easily due to the delimiters serving as a visual anchor.
+* Requiring all patterns to be quoted minimizes the number of characters that need to be escaped within a pattern to 3:
+the 2 pattern delimiter characters and the escape character itself.
+* Because the sum of counts of declarations + `match` statement + `when` statements is always
+greater than or equal to the number of patterns,
+wrapping the entire message once yields less visual noise of repetitive code mode introducer symbols
+when there is 1+ declarations in a `match` (selection) message,
+or when there are 2+ declarations in a non-`match` complex message.
+
+Cons:
+
+* This comes at the cost of an inconsistency in the WYSIWYG patterns are quoted between simple and complex messages.
+In the case of simple messages, the containing format itself implicitly defines the beginning and end of the pattern (example: `"..."`), which is not visible at the level of MF2,
+while complex messages use the aforementioned delimiter to quote patterns (ex: `{{...}}`).
+* Another potential drawback, specifically in the case of non-`match` complex messages with exactly 1 declaration,
+is that this option adds 2 extra delimiters compared to an alternative syntax that doesn't require quoted patterns
+and is designed to minimize delimiter usage only to code mode introducers.
+
+Evaluation:
+
+The pros outweigh the cons, not just in cardinality, but far more importantly, according to the relative weight
+our value system places to the requirements met by the pro aspects compared to the con aspects. Namely:
+
+* [high] Unsurprising WYSIWYG behavior from patterns
+* [high] Easy recognition of patterns, even for non-developers
```

Comment:
To be fair, I've never found ICU MF patterns easy to spot—specifically because they use `{}`.

Reply:
Fair point.

But I've never seen any questions asked for single selections.
(on StackOverflow or internally sites, here and in previous companies)

It gets ugly, even for developers, when you have multiple selections (plural-in-plural, select in plural in ???)